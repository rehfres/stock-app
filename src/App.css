
.App-header {
  height: 40px;
  margin: 2px 5px;
  position: relative;
}


.search-bar {
  width: 100%;
  height: 100%;
  /* max-width: 500px; */
  display: block;
  /* margin: 0; */
  padding: 0 5px;
  border: none;
  box-sizing: border-box;
}

.search-options {
  display: none;
  position: absolute;
  top: calc(100% + 1px);
  left: 0;
  margin: 0;
  padding: 0;
  width: 100%;
  /* max-width: 500px; */
  z-index: 1;
}

.search-bar:focus + .search-options {
  display: unset;
}


.search-options > li {
  height: 40px;
}

.search-options button {
  height: 100%;
  width: 100%;
  display: block;
  background: white;
  border: none;
  cursor: pointer;
  /* padding: 0; */
  text-align: left;
}

.search-options button:hover {
  background: #f0f0f0;
}

.charts-container {
  margin: 10px 10px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}