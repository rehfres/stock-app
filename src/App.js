import React, { Component } from 'react';
// import logo from './logo.svg';
import Stock from './Stock';
import './App.css';
import { auth, getSymbolsFromDb, addSymbolToDb, deleteSymbolFromDb } from './firebase';
// import { Container, Draggable } from 'react-smooth-dnd';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: 'SNAP',
      symbolsListActive: [],
      symbolsList: [],
      searchList: []
    };
    this.search = this.search.bind(this);
  }
  handleClick() {
    console.log(this.state.counter);
    this.setState(state => ({ counter: state.counter + 1 }));
  }
  getSymbolsList() {
    return fetch('https://api.iextrading.com/1.0/ref-data/symbols?filter=symbol')
      .then(response => response.json())
      .then(r => {
        this.setState(state => ({...state, symbolsList: r.map(el => el.symbol)}));
        this.setState(state => ({...state, searchList: state.symbolsList}));
      })
  }
  async getSymbolsListActive() {
    const symbolsListActive = await getSymbolsFromDb();
    console.log('%câ§­', 'color: #2516c7', symbolsListActive);
    this.setState(state => ({...state, symbolsListActive}));
  }
  search(event) {
    const search = (event.target ? event.target.value : event).toUpperCase();
    this.setState(state => ({...state, search}));
    const searchList = this.state.symbolsList.filter(el => el.startsWith(search)) || [];
    this.setState(state => ({...state, searchList: searchList}));
  }
  tryToAddSymbolChart(symbol) {
    if (!this.state.symbolsListActive.some(el => el === symbol)) {
      this.setState(state => ({...state, symbolsListActive: state.symbolsListActive.concat([symbol])}));
      return addSymbolToDb(symbol);
    }
  }
  componentDidMount() {
    auth().then(() => this.getSymbolsListActive());
    this.getSymbolsList().then(() => this.search(this.state.search));
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <input className="search-bar" type="text" placeholder="Search" value={this.state.search} onChange={this.search}/>
          <ul className="search-options">
            {this.state.searchList.map(symbol => <li key={symbol}><button onMouseDown={() => this.tryToAddSymbolChart(symbol)}>{symbol}</button></li>).slice(0, 28)}
          </ul>
        </header>
        <div className="charts-container">
          {this.state.symbolsListActive.map(symbol => <Stock key={symbol} symbol={symbol}/>)}
        </div>
      </div>
    );
  }
}

export default App;
