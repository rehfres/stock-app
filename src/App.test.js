import React from 'react';
import App from './Stock';
import {render, fireEvent, cleanup, waitForElement} from 'react-testing-library';

let getByTestId;

test('renders App', async () => {
  const temp = render(<App />);
  expect(temp).toBeTruthy();
});

// describe('When fetch is working:', () => {
//   beforeAll(() => {
//     cleanup();
//     fetch = jest.fn()
//       .mockImplementationOnce(() => new Promise(resolve => resolve({json: () => [
//         {
//           close: 100,
//           date: "20190208",
//           minute: "09:30",
//         },
//         {
//           close: 200,
//           date: "20190208",
//           minute: "09:31",
//         }
//       ]})))
//       .mockImplementationOnce(() => new Promise(resolve => resolve({json: () => [
//         {
//           close: 50,
//           date: "2019-02-08",
//         }
//       ]})));
//     const dateYesterday = new Date(Date.now() - 864e5).toISOString().split('T')[0].replace(/-/g,'');
//     const dateToday = new Date().getDate();
//     const temp = render(<Stock symbol={'APPL'} dateToday={dateToday} dateYesterday={dateYesterday} canvasSize={{height: 35, width: 100}}/>);
//     getByTestId = temp.getByTestId;
//   })
  
//   test('renders canvas', async () => {
//     const canvas = await waitForElement(() => getByTestId('canvas'));
//     expect(canvas).toBeTruthy();
//   });
  
//   test('renders price', async () => {
//     const price = await waitForElement(() => getByTestId('price'));
//     // console.log('%c⧭', 'color: #00a3cc', price.toHaveTextContent);
//     expect(price).toBeTruthy();
//   });
  
//   test('price is correct', async () => {
//     const price = await waitForElement(() => getByTestId('price'));
//     // console.log('%c⧭', 'color: #00a3cc', price.toHaveTextContent);
//     expect(price).toHaveTextContent('200');
//   });

//   test('renders change', async () => {
//     const change = await waitForElement(() => getByTestId('change'));
//     expect(change).toBeTruthy();
//   });

//   test('change is correct', async () => {
//     const change = await waitForElement(() => getByTestId('change'));
//     expect(change).toHaveTextContent('150 (300%)');
//   });
// })