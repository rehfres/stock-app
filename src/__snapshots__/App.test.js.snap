// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
        style="min-height: 753px;"
      >
        <input
          class="search-bar"
          placeholder="Search"
          type="text"
          value=""
        />
        <ul
          class="search-options"
        />
        <div
          class="charts-container"
        >
          <div
            class="smooth-dnd-container vertical"
          />
        </div>
        <div
          class="button not-signed-in"
        >
          <div
            class="button__logo"
          >
            <img
              alt="Logo"
              src="g-logo.png"
            />
          </div>
          <p
            class="button__text"
          >
            Sign in with Google
          </p>
        </div>
        <div
          class="tip tip--shown"
        >
          <p
            class="tip__text"
          >
            Drag to reorder/remove charts
          </p>
          <img
            alt="close"
            class="close"
            src="close.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
      style="min-height: 753px;"
    >
      <input
        class="search-bar"
        placeholder="Search"
        type="text"
        value=""
      />
      <ul
        class="search-options"
      />
      <div
        class="charts-container"
      >
        <div
          class="smooth-dnd-container vertical"
        />
      </div>
      <div
        class="button not-signed-in"
      >
        <div
          class="button__logo"
        >
          <img
            alt="Logo"
            src="g-logo.png"
          />
        </div>
        <p
          class="button__text"
        >
          Sign in with Google
        </p>
      </div>
      <div
        class="tip tip--shown"
      >
        <p
          class="tip__text"
        >
          Drag to reorder/remove charts
        </p>
        <img
          alt="close"
          class="close"
          src="close.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
